dev/com.ibm.ws.jsp.2.3_fat/fat/src/com/ibm/ws/jsp23/fat/tests/JSTLTests.java

```diff
@@ -0,0 +1,173 @@
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        ShrinkHelper.defaultDropinApp(server, APP_NAME + ".war");
+
+        server.startServer(JSTLTests.class.getSimpleName() + ".log");
+    }
+
+    @AfterClass
+    public static void testCleanup() throws Exception {
+        // Stop the server
+        if (server != null && server.isStarted()) {
+            server.stopServer();
+        }
+    }
+
+    /**
+     * Basic Test for a subset of the Core Tags
+     *
+     * @throws Exception if something goes horribly wrong
+     */
+    @Test
+    public void testCoreTags() throws Exception {
+        WebConversation wc = new WebConversation();
+        wc.setExceptionsThrownOnErrorStatus(false);
+
+        String url = JSPUtils.createHttpUrlString(server, APP_NAME, "coreTags.jsp");
+        LOG.info("url: " + url);
+
+        WebRequest request = new GetMethodWebRequest(url);
+        WebResponse response = wc.getResponse(request);
+        LOG.info("Servlet response : " + response.getText());
+
+
+        assertTrue("#1 Something went wrong with the core tags", response.getText().contains("Item 1"));
+        assertTrue("#2 Something went wrong with the core tags", response.getText().contains("c:when works!"));
+
+    }
+
+    /**
+     * Basic Test for a subset of the Function Tags
+     *
+     * @throws Exception if something goes horribly wrong
+     */
+    @Test
+    public void testFunctionTags() throws Exception {
+        WebConversation wc = new WebConversation();
+        wc.setExceptionsThrownOnErrorStatus(false);
+
+        String url = JSPUtils.createHttpUrlString(server, APP_NAME, "functionTags.jsp");
+        LOG.info("url: " + url);
+
+        WebRequest request = new GetMethodWebRequest(url);
+        WebResponse response = wc.getResponse(request);
+        LOG.info("Servlet response : " + response.getText());
+
+        assertTrue("#1 Something went wrong with function tags", response.getText().contains("Function Tag Works"));
+        assertTrue("#2 Something went wrong with the function tags", response.getText().contains("Length of 'Tags' is: 4"));
+
+    }
+
+    /**
+     * Basic Test for a subset of the XML Tags
+     *
+     * @throws Exception if something goes horribly wrong
+     */
+    @Test
+    public void testXMLTags() throws Exception {
+        WebConversation wc = new WebConversation();
+        wc.setExceptionsThrownOnErrorStatus(false);
+
+        String url = JSPUtils.createHttpUrlString(server, APP_NAME, "xmlTags.jsp");
+        LOG.info("url: " + url);
+
+        WebRequest request = new GetMethodWebRequest(url);
+        WebResponse response = wc.getResponse(request);
+        LOG.info("Servlet response : " + response.getText());
+
+        assertTrue("Something went wrong with the xml tags", response.getText().contains("IBM is trading above 100"));
+
+    }
+
+    /**
+     * Tests Import of SQL Tag Library
+     * 
+     * TODO: Add DB to test tags? 
+     *
```

Comment:
Can you open an issue so we can eventually look into this?

Reply:
Done -- https://github.com/OpenLiberty/open-liberty/issues/16024